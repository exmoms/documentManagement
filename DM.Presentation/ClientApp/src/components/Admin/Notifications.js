import React from "react";
import Typography from "@material-ui/core/Typography";
import { useTranslation } from "react-i18next";
import CloseIcon from "@material-ui/icons/Close";
import { Alert } from "@material-ui/lab";
import IconButton from "@material-ui/core/IconButton";

export function Notification(props) {
  const { t } = useTranslation();
  const { open, error, setOpen, errorMessage } = props;
  return (
    <>
      {open && (
        <Alert
          severity={error ? "error" : "success"}
          action={
            <IconButton
              aria-label="close"
              color="inherit"
              size="small"
              onClick={() => {
                setOpen(false);
              }}
            >
              <CloseIcon fontSize="inherit" />
            </IconButton>
          }
        >
          <Typography variant="subtitle1" align="left" color="primary">
            {error && (
              <ol>
                {errorMessage.map((item, index) => (
                  <li key={index} id={"errorMessageItem"+index}>{item}</li>
                ))}
              </ol>
            )}
            {!error && t("successful_msg")}
          </Typography>
        </Alert>
      )}
    </>
  );
}
