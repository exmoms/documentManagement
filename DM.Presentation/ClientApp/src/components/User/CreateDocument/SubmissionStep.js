/**
 * @file Contains SubmissionStep component implementaion
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @version 1.0
 */
import { Button, CircularProgress, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { Redirect } from "react-router-dom";
import { Notification } from "../../Admin/Notifications";
import CREATE_DOC_STYLES from "../Styles";

const useStyles = makeStyles(CREATE_DOC_STYLES);

const SubmissionStep = (props) => {
  const { t } = useTranslation();
  const classes = useStyles();
  const [redirect, setredirect] = useState(false);
  const [open, setopen] = useState(false);

  const { response, uploading, errors } = props;

  useEffect(() => {
    setopen(response !== null);
  }, [response]);

  switch (uploading) {
    case true:
      return (
        <div>
          <div
            style={{
              margin: "50px",
              padding: "30px 150px",
            }}
          >
            <CircularProgress className={classes.circle} />
          </div>
          <Typography variant="h5" gutterBottom align="center">
            {t("creatdocument_uploadingfilleddoc")}
          </Typography>
        </div>
      );
    case false:
      if (response !== null && response.ok) {
        /* add a button that to create a new doc */
        /* handle submission status */
        return (
          <div>
            <Notification
              id="notifications"
              open={open}
              setOpen={(show) => setopen(show)}
              error={!response.ok}
              errorMessage={errors}
            />
            <div
              style={{
                padding: "30px 150px",
              }}
            >
              <Button
                color="primary"
                onClick={() => {
                  props.handleReset();
                  setredirect(true);
                }}
                className={classes.buttonForm}
              >
                {t("creatdocument_addnewdoc")}
              </Button>
            </div>
            {redirect && <Redirect to="/create-document" />}
          </div>
        );
      } else if (response !== null) {
        return (
          <div>
            <Notification
              id="notifications"
              open={open}
              setOpen={(show) => setopen(show)}
              error={!response.ok}
              errorMessage={errors}
            />
            <div className={classes.buttons}>
              <Button
                color="primary"
                onClick={() => {
                  props.handleReset();
                  setredirect(true);
                }}
                className={classes.buttonForm}
              >
                {t("reset_doc")}
              </Button>
            </div>
            {redirect && <Redirect to="/create-document" />}
          </div>
        );
      }
      break;

    default:
      break;
  }
  return <div></div>;
};

export default SubmissionStep;
