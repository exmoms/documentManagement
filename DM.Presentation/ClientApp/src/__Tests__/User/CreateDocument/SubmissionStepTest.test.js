import React from "react";
import ReactDom from "react-dom";
import SubmissionStep from "../../../components/User/CreateDocument/SubmissionStep.js";
import Notifications from "../../../components/Admin/Notifications.js";
import { cleanup } from "@testing-library/react";
import { mount } from "enzyme";
import Form from "react-jsonschema-form";
import { shallow } from "enzyme";
import { act } from "react-dom/test-utils";
import { Button, Typography, CircularProgress } from "@material-ui/core";
import { Redirect } from "react-router-dom";
afterEach(cleanup);
describe("SubmissionStep", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    let response=null;
    let uploading= true;
    let errorText="";
    const handleResubmit = jest.fn();
    ReactDom.render(
      <SubmissionStep
      response={response}
      uploading={uploading}
      errorText={errorText}
      handleResubmit={handleResubmit}
      />,
      div
    );
  });

  it("matches the snapshot", () => {
    let response=null;
    let uploading= true;
    let errorText="";
    const handleResubmit = jest.fn();
    const wrapper = mount(
        <SubmissionStep
        response={response}
        uploading={uploading}
        errorText={errorText}
        handleResubmit={handleResubmit}
        />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("rendering when uploading is true", () => {
    let response=null;
    let uploading= true;
    let errorText="";
    const handleResubmit = jest.fn();
    const wrapper = shallow(
        <SubmissionStep
        response={response}
        uploading={uploading}
        errorText={errorText}
        handleResubmit={handleResubmit}
        />
    );
   wrapper.update();
   wrapper.render();
   expect(wrapper.find(CircularProgress).length).toBe(1);
   expect(wrapper.find(Typography).text()).toBe("creatdocument_uploadingfilleddoc");
  });
  

  test("rendering when uploading is false and response is ok", () => {
    let response={ok:true};
    let uploading= false;
    let errorText="";
    let errors="errors";
    const consoleMock = jest.fn();
    window.location.reload = consoleMock;
    const handleResubmit = jest.fn();
    const handleReset = jest.fn();
    const wrapper = shallow(
        <SubmissionStep
        response={response}
        uploading={uploading}
        errorText={errorText}
        errors={errors}
        handleResubmit={handleResubmit}
        handleReset={handleReset}
        />
    );
   wrapper.update();
   wrapper.render();
   expect(wrapper.find("#notifications").length).toBe(1);
   expect(wrapper.find("#notifications").at(0).props().open).toBeFalsy();
   expect(wrapper.find("#notifications").at(0).props().error).toBeFalsy();
   expect(wrapper.find("#notifications").at(0).props().errorMessage).toBe(errors);
   expect(wrapper.find(Button).text()).toBe("creatdocument_addnewdoc");
   expect(wrapper.find(Redirect).length).toBe(0);
   //test onSubmit functionality
   act(() => {
    wrapper
      .find(Button)
      .props()
      .onClick();
  });
  wrapper.update();
 //expectation for click event 
  expect(handleReset).toHaveBeenCalledTimes(1);
  expect(wrapper.find(Redirect).length).toBe(1);
  expect(wrapper.find(Redirect).at(0).props().to).toBe("/create-document");
  });
  
  test("rendering when uploading is false and response is not ok", () => {
    let response={ok:false ,status:"failed"};
    let uploading= false;
    let errorText="uncaught error";
    let errors="errors";
    const consoleMock = jest.fn();
    window.location.reload = consoleMock;
    const handleResubmit = jest.fn();
    const handleReset = jest.fn();
    const wrapper = shallow(
        <SubmissionStep
        response={response}
        uploading={uploading}
        errorText={errorText}
        errors={errors}
        handleResubmit={handleResubmit}
        handleReset={handleReset}
        />
    );
   wrapper.update();
   wrapper.render();
   expect(wrapper.find("#notifications").length).toBe(1);
   expect(wrapper.find("#notifications").at(0).props().open).toBeFalsy();
   expect(wrapper.find("#notifications").at(0).props().error).toBeTruthy();
   expect(wrapper.find("#notifications").at(0).props().errorMessage).toBe(errors);
   expect(wrapper.find(Button).text()).toBe("reset_doc");
   expect(wrapper.find(Redirect).length).toBe(0);
   //test setOpen functionality
   act(() => {
    wrapper
      .find("#notifications")
      .props()
      .setOpen(true);
  });
   //test onSubmit functionality
   act(() => {
    wrapper
      .find(Button)
      .props()
      .onClick();
  });
  wrapper.update();
  expect(wrapper.find("#notifications").at(0).props().open).toBeTruthy();
 //expectation for click event 
 expect(wrapper.find(Redirect).length).toBe(1);
  expect(handleReset).toHaveBeenCalledTimes(1);
  expect(wrapper.find(Redirect).at(0).props().to).toBe("/create-document");
  }); 
 
});