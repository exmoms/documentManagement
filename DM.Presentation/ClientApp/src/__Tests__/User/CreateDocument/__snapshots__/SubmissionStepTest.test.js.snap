// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionStep matches the snapshot 1`] = `ReactWrapper {}`;
